DSpace/dspace-angular
# Missing search results statistics

## References
* Fixes [#8729](https://github.com/DSpace/DSpace/issues/8729)
* [PR on REST side](https://github.com/DSpace/DSpace/pull/8796)

## Description
This PR changes the search component to track clicks on listed objects (items/collections/communities) and sends a search event for them, containing the clicked object's UUID so REST can log them as a search_results statistic (see [REST PR](https://github.com/DSpace/DSpace/pull/8796))

## Instructions for Reviewers
**Changes made**
- `SearchService`'s `trackSearch()` method now accepts an optional object UUID
- When present, it'll be added to the search event sent to `angulartics2`
- `Angulartics2DSpace` supports an object property
- Modified `StatisticsService`'s `trackSearchEvent()` to support the object property
- Added a subscribe (which is unsubscribed from on destroy) to the `SearchComponent`, listening to routing events going away from the current page. If it reads one and it's navigating to an object's page (entity, item, collection or community), call `trackSearch()` with the object UUID
- I noticed `SearchTrackerComponent`, our old way of tracking search statistics, still exists and is now completely irrelevant. So I removed it, since we also don't want anyone to re-use it, as it would result in duplicate statistics because it extends from `SearchComponent` itself.

**How to test**
Testing is best done together with a local rest instance containing the changes from the [REST PR](https://github.com/DSpace/DSpace/pull/8796). If not, you can always take a look at the network tab of your browser to spot statistic events being sent.
- Click an item/collection/community from any page but the search page (e.g. homepage), this should _not_ send any statistic events
- Click an item/collection/community from the search page, this should send a statistic event containing the clicked object's UUID
- If using a local REST API with the PR's changes, statistics with type "search_result" should show up with the object's UUID in your solr core.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_
